src/mondeath.cpp

```diff
@@ -721,5 +724,39 @@ void make_mon_corpse(monster *z, int damageLvl)
     if (z->has_effect("no_ammo")) {
         corpse.set_var("no_ammo", "no_ammo");
     }
+    if (corpse.get_mtype()->id==mon_pregnant_giant_cockroach){
```

Comment:
This is hacky. Should be implemented as death functions instead, with `make_mon_corpse` unchanged.
You have death_function: NORMAL for all the roaches. For the pregnant one, you could change it to ["PREG_ROACH", "NORMAL"].
It would leave "pregnant roach corpse", but that's not a big problem. You could try to `poly` the monster inside `preg_roach`.